crates/cheatcodes/src/script.rs

```diff
@@ -65,6 +65,16 @@ impl Cheatcode for signDelegationCall {
     }
 }
 
+impl Cheatcode for signDelegationWithNonceCall {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, privateKey, nonce } = self;
```

Comment:
this is similar with code within `signDelegationCall`, let's have a helper function that takes nonce to avoid redundant code